src/ex.c

```diff
@@ -966,8 +967,38 @@ static VbCmdResult ex_handlers(Client *c, const ExArg *arg)
 
 static VbCmdResult ex_shortcut(Client *c, const ExArg *arg)
 {
-    /* TODO no implemented yet */
-    return CMD_SUCCESS;
+    char *key, *uri;
+    gboolean res = CMD_ERROR;
+
+    if (!c) {
+        return CMD_ERROR;
+    }
+
+    if (!arg || !arg->name) {
+        return CMD_ERROR;
+    }
+
+    if (!arg->rhs || !arg->rhs->str || !*arg->rhs->str) {
+        return CMD_ERROR;
+    }
+
+    if (strstr(arg->name, "shortcut-add")) {
```

Comment:
Each `arg` has the `code` property that can be checked against `EX_SCA`, `EX_SCD` or `EX_SCR` so there is no need to do a string comparison here. I think you can use this function from the master branch except of the new client parameter and the returend values. I'd also prefer not to strip key and uri before adding to the shortcuts until we know that this is required. If there is a need for stripping, we should do this in `shortcut_add` so that possible other callers do not have to concern about stripping as well.